pkg/kubelet/cm: rm some useless code

kolyshkin opened the pull request:
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:

Remove some useless code. See individual commits for details.

#### Which issue(s) this PR fixes:
none

#### Special notes for your reviewer:
none

#### Does this PR introduce a user-facing change?
```release-note
none
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

none

kolyshkin committed:
pkg/kubelet/cm: rm dup code

Commit ecd6361f added setting PidsLimit to Create and Update.

Commit bce9d5f2 added setting PidsLimit to m.toResources.

Now, PidsLimit is assigned twice.

Remove the duplicate.

Fixes: bce9d5f2
Signed-off-by: Kir Kolyshkin <kolyshkin@gmail.com>

odinuge reviewed (changes requested):
This will break all CPU cgroup enforcement between pods and qos classes when using cgroup v2. It is effectively giving all pods the same priority, and that is not something we want.

Is there a special reason that you don't want this logic in k8s in the first place?

/hold

kolyshkin committed:
pkg/kubelet/cm: ToSystemd: nit

Remove the second condition as it can't ever be true (this was obviously
written by a C programmer).

Fixes: b230fb8ac407
Signed-off-by: Kir Kolyshkin <kolyshkin@gmail.com>

kolyshkin commented on pkg/kubelet/cm/cgroup_manager_linux.go:
```diff
@@ -388,11 +377,7 @@ func (m *cgroupManagerImpl) toResources(resourceConfig *ResourceConfig) *libcont
-		} else {
-			resources.CpuShares = *resourceConfig.CpuShares
-		}
+		resources.CpuShares = *resourceConfig.CpuShares
```
You are right; such conversion is performed in `libcontainer/specconv` in runc, but if someone uses cgroup manager directly, they have to convert.

I will remove this hunk.

kolyshkin commented:
OK I guess I'll incorporate this into runc 1.1 PR
